### PR TITLE
Update ffiPaymentRedirect to use compute

### PR DESCRIPTION
**Pull request by atavism**
- Update ffiPaymentRedirect to use compute
- Add PaymentRedirect API response type
- Update email exists check to return string instead of pointer

This also fixes an issue on main with desktop where the following error is shown after opening the payment redirect page

Screenshot 2024-04-10 at 3 41 32 PM

### CODE REVIEW

**Review by jigar-f — APPROVED** [review]
LGTM!

File: lib/plans/checkout.dart [hunk 1]
```diff
@@ -409,7 +406,7 @@ class _CheckoutState extends State<Checkout>
                           text: 'continue'.i18n,
                           // for Pro users renewing their accounts, we always have an e-mail address
                           // so it's unnecessary to disable the continue button
-                          disabled: !widget.isPro ? !showContinueButton : false,
+                          disabled: !enableContinueButton(),
```
Comment (jigar-f) [inline lib/plans/checkout.dart:409]: @atavism  I was able to click the continue button without email so, I changed to use !enableContinueButton()

File: lib/plans/checkout.dart [hunk 2]
```diff
@@ -69,8 +69,6 @@ class _CheckoutState extends State<Checkout>
 
   @override
   void dispose() {
-    emailController.dispose();
-    refCodeController.dispose();
```
Comment (jigar-f) [inline lib/plans/checkout.dart:73]: We are already doing dispose on CTextField, Doing it twice was throwing error.

File: lib/plans/checkout.dart [hunk 3]
```diff
@@ -29,7 +29,7 @@ class _CheckoutState extends State<Checkout>
   final emailFieldKey = GlobalKey<FormState>();
   late final emailController = CustomTextEditingController(
     formKey: emailFieldKey,
-    validator: (value) => EmailValidator.validate(value ?? '')
+    validator: (value) => value!.isEmpty?null: EmailValidator.validate(value ?? '')
```
Comment (jigar-f) [inline lib/plans/checkout.dart:32]: It was showing an error before even entering the email, I changed it to when the user starts to enter then we threw an error.

**Comment by jigar-f** [comment]
I am good with this @atavism, Merging...

File: lib/plans/checkout.dart (hunk 2 above)
Comment (atavism) [reply lib/plans/checkout.dart:73]: Thanks for fixing this!

**Comment by atavism** [comment]
Thanks for merging and the review, @jigar-f!

**Pull request merged**